packages/lab/src/__tests__/__e2e__/date-picker/DatePicker.cy.tsx

```diff
@@ -0,0 +1,205 @@
+  DateValue,
+  getLocalTimeZone,
+  startOfMonth,
+  today,
+} from "@internationalized/date";
+
+const composedStories = composeStories(datePickerStories);
+const { Default, Range } = composedStories;
+
+const testDate = new CalendarDate(2000, 2, 1);
+const testInput = "02-feb-2000";
+const rangeTestInput = "02-mar-2000";
+const localTimeZone = getLocalTimeZone();
+const currentLocale = navigator.languages[0];
+const formatDate = (date: DateValue, options?: Intl.DateTimeFormatOptions) => {
+  const formatter = new DateFormatter(currentLocale, options);
+  return formatter.format(date.toDate(localTimeZone));
+};
+const formatInput = (date: DateValue): string =>
+  new DateFormatter("EN-GB", {
+    day: "2-digit",
+    month: "short",
+    year: "numeric",
+  }).format(date.toDate(localTimeZone));
+
+const formatDay = (date: DateValue) => {
+  return formatDate(date, {
+    day: "2-digit",
+    month: "long",
+    year: "numeric",
+  });
+};
+
+describe("GIVEN a DatePicker", () => {
+  checkAccessibility(composedStories);
+
+  describe("WHEN single datepicker is mounted", () => {
+    it("THEN it should mount with the specified defaultStartDate", () => {
+      cy.mount(<Default defaultStartDate={testDate} />);
+      cy.findByRole("textbox").should("have.value", formatInput(testDate));
+    });
+    it("THEN should format a valid date with a different format on blur", () => {
+      const onChange = (event: ChangeEvent<HTMLInputElement>) => {
+        // React 16 backwards compatibility
+        event.persist();
+      };
+      cy.mount(<Default defaultStartDate={testDate} onChange={onChange} />);
+      cy.findByRole("textbox").click().clear().type(testInput);
+      cy.findByRole("textbox").blur();
+      cy.findByRole("textbox").should(
+        "have.value",
+        formatInput(testDate.add({ days: 1 }))
+      );
+    });
+    it("THEN should error and not format invalid dates on blur", () => {
+      cy.mount(<Default defaultStartDate={testDate} />);
+      cy.findByRole("textbox").click().clear().type("date");
+      cy.findByRole("textbox").blur();
+      cy.findByRole("textbox").should("have.value", "date");
+    });
+    it("THEN clicking the calendar button should open the panel", () => {
+      cy.mount(<Default defaultStartDate={testDate} />);
+      cy.findByRole("button", { name: "Open Calendar" }).realClick();
+      cy.findByRole("application").should("exist");
+    });
+    it("THEN should close the calendar panel once a date is selected", () => {
+      cy.mount(<Default defaultStartDate={testDate} />);
+      cy.findByRole("button", { name: "Open Calendar" }).realClick();
+      cy.findByRole("button", {
+        name: formatDay(testDate.add({ days: 11 })),
+      }).should("exist");
+      cy.findByRole("button", {
+        name: formatDay(testDate.add({ days: 11 })),
+      }).realClick();
+      cy.findByRole("application").should("not.exist");
+    });
+    it("THEN should not open the calendar when disabled", () => {
```

Comment:
Description doesn't really match the test? This feels more like "open button is disabled"